x-pack/plugins/siem/public/components/navigation/index.test.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { shallow } from 'enzyme';
+import { LocationDescriptorObject, LocationListener, UnregisterCallback } from 'history';
+import * as React from 'react';
+
+import { SiemNavigationComponent } from './';
+import { setBreadcrumbs } from './breadcrumbs';
+
+jest.mock('./breadcrumbs', () => ({
+  setBreadcrumbs: jest.fn(),
+}));
+type Action = 'PUSH' | 'POP' | 'REPLACE';
+const pop: Action = 'POP';
+describe('SIEM Navigation', () => {
+  const location = {
+    pathname: '/hosts',
+    search: '',
+    state: '',
+    hash: '',
+  };
+  const mockProps = {
+    location,
+    match: {
+      isExact: true,
+      params: {},
+      path: '',
+      url: '',
+    },
+    history: {
+      length: 2,
+      location,
+      action: pop,
+      push: () => null,
+      replace: () => null,
+      go: () => null,
+      goBack: () => null,
+      goForward: () => null,
+      block: (t: UnregisterCallback) => t,
+      createHref: (t: LocationDescriptorObject) => 't',
+      listen: (l: LocationListener) => {
+        const temp: UnregisterCallback = () => null;
+        return temp;
+      },
+    },
+  };
+  const wrapper = shallow(<SiemNavigationComponent {...mockProps} />);
+  test('it calls setBreadcrumbs with correct path on mount', () => {
+    // @ts-ignore property mock does not exists
+    expect(setBreadcrumbs.mock.calls[0][0]).toEqual('/hosts');
```

Comment:
if anyone knows a better way to get past this ts err?

Reply:
What about `.toHaveBeenNthCalledWith(nthCall, arg1, arg2, ....)` ?
https://jestjs.io/docs/en/expect#tohavebeennthcalledwithnthcall-arg1-arg2-
We could bypass accessing mock by doing this?

Reply:
thanks @angorayc!

Reply:
@angorayc That's a great way of doing it!!! Much cleaner